File: 209-minimum-size-subarray-sum/209-minimum-size-subarray-sum.py
class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        res = float('inf')
        l = 0
        count = 0
        for r in range(len(nums)):
            count += nums[r]
                
            while count >= target and l <= r:
                res = min(res, r - l + 1)
                count -= nums[l]
                l += 1
                
        return res if res < float('inf') else 0

File: 209-minimum-size-subarray-sum/209-minimum-size-subarray-sum.java
class Solution {
    public int minSubArrayLen(int target, int[] nums) {
        int res = Integer.MAX_VALUE;
        int l = 0;
        int count = 0;
        for(int r = 0; r < nums.length; r++){
            count += nums[r];
            while(count >= target && l <= r){
                int temp = Math.min(r - l + 1, res);
                res = temp;
                count -= nums[l];
                l += 1;
            }
        }
        if(res < Integer.MAX_VALUE){
            return res;
        }
        return 0;
        
    }
}